File: javascript/isLucky.js
function solution(n) {
  numberArray = Array.from(String(n), Number);

  const middleIndex = Math.floor(numberArray.length / 2);

  const firstArray = numberArray.slice(0, middleIndex);
  const secondArray = numberArray.slice(middleIndex);

  const firstArraySum = sumArray(firstArray);
  const secondArraySum = sumArray(secondArray);

  return firstArraySum === secondArraySum;
}

function sumArray(array) {
  return array.reduce(
    (accumulator, currentValue) => accumulator + currentValue,
    0
  );
}

File: ruby/is_lucky.rb
def solution(n)
  array_number = n.to_s.split('')
  middle_index = array_number.size / 2

  first_part = array_number.slice(0, middle_index)
  second_part = array_number.slice(middle_index, array_number.size)

  sum_array(first_part) == sum_array(second_part)
end

def sum_array(array)
  array.map(&:to_i).reduce(:+)
end
